wp-content/themes/core/components/blocks/hero/Controller.php

```diff
@@ -61,61 +60,93 @@ class Controller extends Abstract_Controller {
+			$args[ $key ] = array_merge( $args[ $key ], $value );
+		}
+
+		$this->layout            = $args[ 'layout' ];
+		$this->media             = $this->get_image( $args[ 'media' ] );
+		$this->content           = $this->get_content( $args[ 'content' ] );
+		$this->container_classes = (array) $args[ 'container_classes' ];
+		$this->media_classes     = (array) $args[ 'media_classes' ];
+		$this->content_classes   = (array) $args[ 'content_classes' ];
+		$this->classes           = (array) $args[ 'classes' ];
+		$this->attrs             = (array) $args[ 'attrs' ];
+		$this->init();
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function defaults(): array {
+		return [
+			'layout'            => Hero_Block::LAYOUT_LEFT,
+			'media'             => [],
+			'content'           => [],
+			'container_classes' => [ 'b-hero__container', 'l-container' ],
+			'media_classes'     => [ 'b-hero__media' ],
+			'content_classes'   => [ 'b-hero__content' ],
+			'classes'           => [
 				'c-block',
 				'b-hero',
 				'c-block--full-bleed',
-				'c-block--' . $this->layout,
-			] );
-		$this->attrs             = (array) ( $args[ 'attrs' ] ?? [] );
+			],
+			'attrs'             => [],
+		];
 	}
 
 	/**
-	 * @param array $args
-	 *
 	 * @return array
 	 */
-	protected function get_content( $args ): array {
-		if ( ! isset( $args[ 'content' ] ) || empty( $args[ 'content' ] ) ) {
-			return [];
-		}
+	protected function required(): array {
+		return [];
+	}
 
-		return [
-			'classes' => [ 'b-hero__content-container', 't-theme--light' ],
-			'leadin'  => $args[ 'content' ][ Model::LEAD_IN ] ?? '',
-			'title'   => $args[ 'content' ][ Model::TITLE ] ?? '',
-			'text'    => $args[ 'content' ][ Model::TEXT ] ?? '',
-			'action'  => $args[ 'content' ][ Model::CTA ] ?? [],
-			'layout'  => $this->layout,
-		];
+	/**
+	 * Any required setup stuff
+	 */
+	protected function init() {
+		$this->classes[] = 'c-block--' . $this->layout;
 	}
 
 	/**
 	 * @param array $args
 	 *
-	 * @return array
+	 * @return string
 	 */
-	protected function get_media( $args ): array {
-		if ( ! isset( $args[ 'media' ] ) ) {
-			return [];
+	protected function get_content( $content ): string {
+		if ( empty( $content ) ) {
+			return '';
 		}
 
-		return $this->get_image( $args[ 'media' ] );
+		return tribe_template_part( 'components/content_block/content_block', null, [
+			'classes' => [ 'b-hero__content-container', 't-theme--light' ],
+			'leadin'  => defer_template_part( 'components/text/text', null, [
+				'content' => $content[ Hero::LEAD_IN ] ?? '',
+			] ),
+			'title'   => defer_template_part( 'components/text/text', null, [
+				'content' => $content[ Hero::TITLE ] ?? '',
+			] ),
+			'content' => defer_template_part( 'components/container/container', null, [
+				'content' => $content[ Hero::DESCRIPTION ] ?? '',
+			] ),
+			'cta'  => defer_template_part( 'components/link/link', null, $content[ Hero::CTA ] ?? [] ),
+			'layout'  => $this->layout,
+		] );
```

Comment:
@msawicki thinking...give we want these block patterns as components to be usable outside of the context of ACF, we probably don't want to be baking in ACF field names here right?